src/main/scala/Validator.scala

```diff
@@ -12,10 +12,21 @@ import scala.collection.mutable
 import scala.collection.mutable.{Map, Set}
 import scala.jdk.CollectionConverters.{IterableHasAsScala, MapHasAsScala}
 
-class Validator(var tableCsvFile: URI, sourceUri: String = "") {
+class Validator(var schemaUri: String, sourceUri: String = "") {
   val mapAvailableCharsets = Charset.availableCharsets().asScala
+  private def getAbsoluteSchemaUri(schemaPath: String): URI = {
+    val inputSchemaUri = new URI(schemaPath)
+    if (inputSchemaUri.getScheme == null) {
+      new URI(s"file://${new File(schemaPath).getAbsolutePath}")
+    } else {
+      inputSchemaUri
+    }
+  }
+
   def validate(): WarningsAndErrors = {
-    Schema.loadMetadataAndValidate(tableCsvFile) match {
+    if (schemaUri.isEmpty) return WarningsAndErrors()
```

Comment:
I'm a bit worried that this will conflict with the code which fetches a CSV from the web and is given the schema location in a header. At the least can you put a comment above here which warns that we need to do HTTP fetching of the CSV before this point (and explain why)?